python_modules/dagster/dagster/_core/storage/alembic/versions/047_add_run_tags_run_id_idx.py

```diff
@@ -0,0 +1,54 @@
+"""add run_id to run_tags index
+
+Revision ID: 6b7fb194ff9c
+Revises: 16e3655b4d9b
+Create Date: 2024-11-14 22:19:46.456731
+
+"""
+
+from alembic import op
+from dagster._core.storage.migration.utils import has_index, has_table
+
+# revision identifiers, used by Alembic.
+revision = "6b7fb194ff9c"
+down_revision = "16e3655b4d9b"
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    if has_table("run_tags"):
+        if not has_index("run_tags", "idx_run_tags_run_id"):
+            op.create_index(
```

Comment:
Is there a better approach than creating new index first and  dropping the old one after?
Do you know if it will run sequentially? I am unsure if the  `postgresql_concurrently=True` won't make it unblocking. In that case, it would drop the old index before the new one would be finished.
I would like to make this as little affecting users as possible (thought it is recommended to downscale controller/UI first).

Reply:
This should be safe.  `postgresql_concurrently` means that it won't lock the table during index creation, but the alembic operations should happen serially.